src/Substructure.jl

```diff
@@ -0,0 +1,313 @@
+"""
+    has_parents(p, history) -> (boolean, Int64, Int64)
+
+Checks if the jet `p` is a child of two other jets, after clustering 
+
+# Arguments
+- `p`: The jet to check.
+- `history`: The vector of history element obtained from the cluster sequence after clustering.
+
+# Returns
+- (boolean, Int64, Int64): true or false depending on if the jet has a parent or not. If the jet has a parent, returns the indices of the parent jets in the history element. Otherwise, returns -2 (NonexistentParent).
+"""
+function has_parents(p::PseudoJet, history::Vector{HistoryElement})
+    
+    N = p._cluster_hist_index
+    p1 = history[N].parent1
+    p2 = history[N].parent2
+
+    p1 == p2 == NonexistentParent ? result = false : result = true
+    return (result, p1, p2)
+
+end
+
+"""
+    delta_R(jet1, jet2) -> Float64
+
+Function to calculate the distance in the y-ϕ plane between two jets `jet1` and `jet2`
+
+# Arguments
+- `jet1`: The first jet.
+- `jet2`: The second jet.
+
+# Returns
+- Float64: The Euclidean distance in the y-ϕ plane for the two jets.
+"""
+function delta_R(jet1::PseudoJet, jet2::PseudoJet)
+    
+    eta1, phi1 = rapidity(jet1), phi(jet1)
+    eta2, phi2 = rapidity(jet2), phi(jet2)
+
+    d_eta = eta1 - eta2
+    d_phi = phi1 - phi2
+    d_phi = abs(d_phi) > π ? 2π - abs(d_phi) : d_phi
+    
+    return sqrt(d_eta^2 + d_phi^2)
+
+end
+
+# Function to calculate kt distance between two pseudojets
+function kt_distance(jet1::PseudoJet, jet2::PseudoJet, R = 1)
+    p1 = pt2(jet1)
+    p2 = pt2(jet2)
+    
+    d_R = delta_R(jet1, jet2)
+    return min(p1, p2) * (d_R^2 / R^2)
+end;
+
+"""
+    recluster(jet, clusterseq, rad, mtd) -> ClusterSequence
+
+Reclusters the constituents of a given jet `jet` with a different clustering method `mtd` and different jet radius `rad`.
+
+# Arguments
+- `jet`: The jet whose constituents are to be reclustered.
+- `clusterseq`: The cluster sequence from which the original jet is obtained.
+- `rad`: The new jet radius.
+- `mtd`: The new clustering method.
+
+# Returns
+- ClusterSequence: The new cluster sequence.
+"""
+function recluster(jet::PseudoJet, clusterseq::ClusterSequence, rad = 1.0, mtd = 0)
+    cons = constituents(jet, clusterseq)
+    new_clusterseq = jet_reconstruct(cons; p = mtd, R = rad, strategy = RecoStrategy.N2Plain)
+   
+    return new_clusterseq
+end
+
+function sort_jets!(event_jet_array)
+    jet_pt(jet) = pt2(jet)
+    sort!(event_jet_array, by = jet_pt, rev = true)
+    # println(event_jet_array)
+end
+
+function join(jets::Vector{PseudoJet})
+    final = PseudoJet(0.0, 0.0, 0.0, 0.0)
+    
+    for jet in jets
```

Comment:
You can replace with the more Julia idiomatic:

```Julia
foldl(+, jets)
```

Then you don't even need `final` I think.

In fact it probably doesn't even need the function wrapper now!